packages/backend-common/src/reading/AwsS3UrlReader.ts

```diff
@@ -42,20 +48,45 @@ const parseURL = (
    */
   pathname = pathname.substr(1);
 
+  let bucket;
+  let region;
+
   /**
-   * Checks that the given URL is a valid S3 object url.
-   * Format of a Valid S3 URL: https://bucket-name.s3.Region.amazonaws.com/keyname
+   * Path style URLs: https://s3.Region.amazonaws.com/bucket-name/key-name
+   * Virtual hosted style URLs: https://bucket-name.s3.Region.amazonaws.com/key-name
+   * See https://docs.aws.amazon.com/AmazonS3/latest/userguide/VirtualHosting.html#path-style-access
    */
-  const validHost = new RegExp(
-    /^[a-z\d][a-z\d\.-]{1,61}[a-z\d]\.s3\.[a-z\d-]+\.amazonaws.com$/,
-  );
-  if (!validHost.test(host)) {
-    throw new Error(`not a valid AWS S3 URL: ${url}`);
+  if (config.s3ForcePathStyle) {
+    if (pathname.indexOf('/') < 0) {
+      throw new Error(
+        `invalid path-style AWS S3 URL, ${url} does not contain bucket in the path`,
+      );
+    }
+    [bucket] = pathname.split('/');
+    pathname = pathname.substr(bucket.length + 1);
+  } else {
+    if (host.indexOf('.') < 0) {
+      throw new Error(
+        `invalid virtual hosted-style AWS S3 URL, ${url} does not contain bucket prefix in the host`,
+      );
+    }
+    [bucket] = host.split('.');
+    host = host.substr(bucket.length + 1);
   }
 
-  const [bucket] = host.split(/\.s3\.[a-z\d-]+\.amazonaws.com/);
-  host = host.substring(bucket.length);
-  const [, , region, ,] = host.split('.');
+  // Only extract region from *.amazonaws.com hosts
+  if (config.host === AMAZON_AWS_HOST) {
+    // At this point bucket prefix is removed from host for virtual hosted URLs
+    const match = host.match(/^s3\.([a-z\d-]+)\.amazonaws.com$/);
```

Comment:
Could be worth fixing the missing `\.com` as part of this

Reply:
fixed